compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -1972,7 +1972,7 @@ public void visit(BLangSimpleVariable varNode) {
         if (varSymbol.type.tag == TypeTags.INVOKABLE) {
             BInvokableSymbol symbol = (BInvokableSymbol) varSymbol;
             BInvokableTypeSymbol tsymbol = (BInvokableTypeSymbol) symbol.type.tsymbol;
-            symbol.params = tsymbol.params;
+            symbol.params = tsymbol.params == null ? null : new ArrayList<>(tsymbol.params);
```

Comment:
`param` list can be null only with a function without a signature. Isn't it?

Reply:
Yes, but I don't see the difference between checking for null vs checking for a specific flag. Can we do it better?

Reply:
The other thing we can do is check the flag and set the `rest param`, `return type` and `param list` as null. But this way is okay for me.